React/Views/RCTNavItemManager.m

```diff
@@ -39,4 +39,21 @@ - (UIView *)view
 RCT_EXPORT_VIEW_PROPERTY(rightButtonIcon, UIImage)
 RCT_EXPORT_VIEW_PROPERTY(rightButtonTitle, NSString)
 
+RCT_CUSTOM_VIEW_PROPERTY(titleFontSize, CGFloat, RCTNavItem)
+{
+  view.titleFont = [RCTConvert UIFont:view.titleFont withSize:json ?: @(defaultView.titleFont.pointSize)];
+}
+RCT_CUSTOM_VIEW_PROPERTY(titleFontWeight, NSString, RCTNavItem)
+{
+  view.titleFont = [RCTConvert UIFont:view.titleFont withWeight:json]; // defaults to normal
```

Comment:
I believe if `json` is nil, the weight just doesn't change if `view.titleFont` already has a custom weight but it should get reset to the default value here (like you did for the font size).